SharkCage/CageManager/CageManager.cpp

```diff
@@ -539,3 +547,51 @@ std::optional<std::wstring> generateUuid() {
 	::RpcStringFree(&uuid_str);
 	return uuid_stl;
 }
+
+void quitChildProcesses(std::wstring process_binary_name) 
+{
+	DWORD my_pid = GetCurrentProcessId();
+	HANDLE snapshot_handle = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS,0);
+	PROCESSENTRY32 process_entry;
+	process_entry.dwSize = sizeof(PROCESSENTRY32);
+
+	if (snapshot_handle == INVALID_HANDLE_VALUE)
+	{
+		std::wcout << L"Cannot terminate ctfmon.exe - cannot enumerate processes" << std::endl;
+		return;
+	}
+
+	if (!Process32First(snapshot_handle, &process_entry))
+	{
+		std::wcout << "Cannot terminate ctfmon.exe - canot read snapshot" << std::endl;
+		CloseHandle(snapshot_handle);
+		return;
+	}
+
+	do
+	{
+		//is child of CageManager?
+		if (process_entry.th32ParentProcessID == my_pid)
+		{
+			//is proper name?
+			if (wcscmp(process_entry.szExeFile, process_binary_name.c_str()) == 0)
+			{
+				//terminate process
+				HANDLE process_handle = OpenProcess(PROCESS_TERMINATE, FALSE, process_entry.th32ProcessID);
+				if (process_handle == NULL)
```

Comment:
nullptr ;)

Reply:
This was intentional, documentation to `OpenProcess` states `NULL` as an error return value... I am not sure whether it is guaranteed that `NULL` as defined in winapi (since it is a define) and `nullptr` are equivalent?

Reply:
Raymond Chen (longtime MS employee) says they are equivalent:
> Therefore, you are welcome to write nullptr instead of NULL if you're writing C++ code. You're also welcome to write anything else that produces a null pointer [...]

https://blogs.msdn.microsoft.com/oldnewthing/20180307-00/?p=98175

And the MSDN doc for nullptr (https://docs.microsoft.com/en-gb/cpp/cpp/nullptr) recommends using it instead of NULL:
> Avoid using NULL or zero (0) as a null pointer constant; nullptr is less vulnerable to misuse and works better in most situations. 

Personally I've used it for a few years now while working with Windows (API) specific code and never had any problems :)